src/Compilers/Core/Portable/Compilation/SemanticModel.cs

```diff
@@ -91,7 +91,27 @@ public IOperation GetOperation(SyntaxNode node, CancellationToken cancellationTo
         /// </summary>
         public ImmutableArray<Operations.BasicBlock> GetControlFlowGraph(Operations.IBlockOperation body)
         {
-            return Operations.ControlFlowGraphBuilder.Create(body);
+            if (body == null)
+            {
+                throw new ArgumentNullException(nameof(body));
+            }
+
+            if (!body.Syntax.SyntaxTree.Options.Features.ContainsKey("flow-analysis"))
+            {
+                throw new InvalidOperationException(CodeAnalysisResources.FlowAnalysisFeatureDisabled);
+            }
+
+            try
+            {
+                return Operations.ControlFlowGraphBuilder.Create(body);
```

Comment:
so, control flow doesn't actually require semantic model to build? then is there reason why we didn't want to put it in

body.GetControlFlowGraph()?

Reply:
@AlekseyTs @jinujoseph @gafter